File: php/features.php
<?php
declare(strict_types=1);

// BluefinDecryptxP2pe SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/AuditFeature.php';
require_once __DIR__ . '/feature/ClienttrackFeature.php';
require_once __DIR__ . '/feature/IdempotencyFeature.php';
require_once __DIR__ . '/feature/LogFeature.php';
require_once __DIR__ . '/feature/MetricsFeature.php';
require_once __DIR__ . '/feature/PagingFeature.php';
require_once __DIR__ . '/feature/RatelimitFeature.php';
require_once __DIR__ . '/feature/RetryFeature.php';
require_once __DIR__ . '/feature/TelemetryFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';
require_once __DIR__ . '/feature/TimeoutFeature.php';


class BluefinDecryptxP2peFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new BluefinDecryptxP2peBaseFeature();
            case "audit":
                return new BluefinDecryptxP2peAuditFeature();
            case "clienttrack":
                return new BluefinDecryptxP2peClienttrackFeature();
            case "idempotency":
                return new BluefinDecryptxP2peIdempotencyFeature();
            case "log":
                return new BluefinDecryptxP2peLogFeature();
            case "metrics":
                return new BluefinDecryptxP2peMetricsFeature();
            case "paging":
                return new BluefinDecryptxP2pePagingFeature();
            case "ratelimit":
                return new BluefinDecryptxP2peRatelimitFeature();
            case "retry":
                return new BluefinDecryptxP2peRetryFeature();
            case "telemetry":
                return new BluefinDecryptxP2peTelemetryFeature();
            case "test":
                return new BluefinDecryptxP2peTestFeature();
            case "timeout":
                return new BluefinDecryptxP2peTimeoutFeature();
            default:
                return new BluefinDecryptxP2peBaseFeature();
        }
    }

    /**
     * Does a generated feature class back this name? False for a name only
     * an options extend instance can supply (the station adopt path) - the
     * constructor uses this to skip make_feature for such names instead of
     * adding a stray BaseFeature.
     */
    public static function has_feature(string $name): bool
    {
        switch ($name) {
            case "base":
            case "audit":
            case "clienttrack":
            case "idempotency":
            case "log":
            case "metrics":
            case "paging":
            case "ratelimit":
            case "retry":
            case "telemetry":
            case "test":
            case "timeout":
                return true;
            default:
                return false;
        }
    }
}

File: rb/features.rb
# BluefinDecryptxP2pe SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/audit_feature'
require_relative 'feature/clienttrack_feature'
require_relative 'feature/idempotency_feature'
require_relative 'feature/log_feature'
require_relative 'feature/metrics_feature'
require_relative 'feature/paging_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/telemetry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module BluefinDecryptxP2peFeatures
  def self.make_feature(name)
    case name
    when "base"
      BluefinDecryptxP2peBaseFeature.new
    when "audit"
      BluefinDecryptxP2peAuditFeature.new
    when "clienttrack"
      BluefinDecryptxP2peClienttrackFeature.new
    when "idempotency"
      BluefinDecryptxP2peIdempotencyFeature.new
    when "log"
      BluefinDecryptxP2peLogFeature.new
    when "metrics"
      BluefinDecryptxP2peMetricsFeature.new
    when "paging"
      BluefinDecryptxP2pePagingFeature.new
    when "ratelimit"
      BluefinDecryptxP2peRatelimitFeature.new
    when "retry"
      BluefinDecryptxP2peRetryFeature.new
    when "telemetry"
      BluefinDecryptxP2peTelemetryFeature.new
    when "test"
      BluefinDecryptxP2peTestFeature.new
    when "timeout"
      BluefinDecryptxP2peTimeoutFeature.new
    else
      BluefinDecryptxP2peBaseFeature.new
    end
  end
end
